File: 2025/day06_1/src/main/kotlin/Main.kt
package org.tlarsen.adventofcode2025

import java.io.File

fun main(argv: Array<String>) {
    println("AOC 2025, Day 6, Part 1 starting!!!!")
    println("Details of problem are here: https://adventofcode.com/2025/day/6#part1")

    val allFormulas = mutableListOf<CephalopodFormula>()
    var firstLine = true

    File(argv[0]).forEachLine {
        val theSplit = it.split(regex = "\\s+".toRegex()).filter { its -> its.isNotBlank() }
        var splitCounter = 0
        for(split in theSplit) {
            if(split.isNotEmpty()) {
                if (firstLine) {
                    // special parsing for the first line: use it to initialize array list.
                    val newFormula = CephalopodFormula()
                    newFormula.addParam(split.toInt())
                    allFormulas.add(newFormula)
                } else if((split == "+") || (split == "*")) {
                    // special parsing operator
                    val formula = allFormulas[splitCounter]
                    when (split) {
                        "+" -> {
                            formula.operation = CephalopodOperation.ADDITION
                        }
                        "*" -> {
                            formula.operation = CephalopodOperation.MULTIPLICATION
                        }
                        else -> {
                            println("WARN: Could not parse operator: $split")
                        }
                    }
                } else {
                    // normal number parsing
                    val formula = allFormulas[splitCounter]
                    formula.addParam(split.toInt())
                }
                splitCounter++
            }
        }
        firstLine = false
    }
    var theTotal = 0L
    for(formula in allFormulas) {
        val answer = formula.doHomeworkNew()
        println("Formula: $formula".padEnd(30, ' ') + " = $answer")
        theTotal += answer
    }

    println("The total calculated using Cephalopod Math: $theTotal")

    println("AOC 2025, Day 6, Part 1 completed!!!")
}

File: 2025/day06_1/src/main/python/main.py
import sys
from cephalopod_formula import CephalopodFormula
from cephalopod_operation import CephalopodOperation


def main():
    print("AOC 2025, Day 6, Part 1 starting!!!!")
    print("Details of problem are here: https://adventofcode.com/2025/day/6#part1")

    all_formulas = []
    first_line = True

    with open(sys.argv[1]) as f:
        for line in f:
            the_split = line.split()
            split_counter = 0
            for split in the_split:
                if not split:
                    continue

                if first_line:
                    new_formula = CephalopodFormula()
                    new_formula.add_param(int(split))
                    all_formulas.append(new_formula)
                elif split == "+" or split == "*":
                    formula = all_formulas[split_counter]
                    if split == "+":
                        formula.operation = CephalopodOperation.ADDITION
                    elif split == "*":
                        formula.operation = CephalopodOperation.MULTIPLICATION
                else:
                    formula = all_formulas[split_counter]
                    formula.add_param(int(split))

                split_counter += 1

            first_line = False

    total = 0
    for formula in all_formulas:
        answer = formula.do_homework()
        print(f"Formula: {str(formula).ljust(30)} = {answer}")
        total += answer

    print(f"The total calculated using Cephalopod Math: {total}")

    print("AOC 2025, Day 6, Part 1 completed!!!")


if __name__ == "__main__":
    main()